packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,293 @@
+      address = res.flags.validator
+      await newCheckBuilder(this)
+        .isValidator(address)
+        .runChecks()
+    }
+
+    if (res.flags.group) {
+      if (address) throw Error('Must select --voter, --validator, or --group')
+      address = res.flags.group
+      await newCheckBuilder(this)
+        .isValidatorGroup(address)
+        .runChecks()
+    }
+
+    // For each epoch...
+    for (let blockNumber = fromBlock; blockNumber <= lastEpochBlock; blockNumber += epochSize) {
+      // Get the groups that address voted for at this epoch, and those groups' total votes.
+      const activeAddressVotes: { [key: string]: BigNumber } = {}
+      const activeGroupVotes: { [key: string]: BigNumber } = {}
+      if (address) {
+        const voter = await election.getVoter(address, blockNumber)
+        for (const vote of voter.votes) {
+          const group: string = vote.group.toLowerCase()
+          activeAddressVotes[group] = (activeAddressVotes[group] || new BigNumber(0)).plus(
+            vote.active
+          )
+          activeGroupVotes[group] = (activeGroupVotes[group] || new BigNumber(0)).plus(
+            await election.getTotalVotesForGroup(group, blockNumber)
+          )
+        }
+      }
+
+      // voterRewardsEvent applies to address when voterRewardsEvent.group in activeAddressVotes.
+      const epochVoterRewardsEvents = await election.getPastVoterRewards({
+        fromBlock: blockNumber,
+        toBlock: blockNumber,
+      })
+
+      const voterRewardsValidatorGroupDetails: ValidatorGroup[] = await concurrentMap(
+        10,
+        epochVoterRewardsEvents,
+        (e: EventLog) => validators.getValidatorGroup(e.returnValues.group)
+      )
+
+      const epochVoterRewards = epochVoterRewardsEvents.map(
+        (e: EventLog, index: number): VoterReward => ({
+          group: e.returnValues.group,
+          groupName: voterRewardsValidatorGroupDetails[index].name,
+          totalVoterPayment: e.returnValues.value,
+          blockNumber: e.blockNumber,
+          activeAddressVotes: activeAddressVotes[e.returnValues.group],
+          activeGroupVotes: activeGroupVotes[e.returnValues.group],
+        })
+      )
+
+      voterRewards = voterRewards.concat(
+        address
+          ? epochVoterRewards.filter(
+              (e: VoterReward) => e.group.toLowerCase() in activeAddressVotes
+            )
+          : epochVoterRewards
+      )
+
+      // validatorRewardEvent applies to address when validatorRewardEvent.validator is address.
+      const epochValidatorRewardsEvents = await validators.getPastValidatorRewards({
+        fromBlock: blockNumber,
+        toBlock: blockNumber,
+      })
+
+      const epochValidatorDetails: Validator[] = await concurrentMap(
+        10,
+        epochValidatorRewardsEvents,
+        (e: EventLog) => validators.getValidator(e.returnValues.validator, blockNumber)
+      )
+
+      const epochValidatorRewards = epochValidatorRewardsEvents.map(
```

Comment:
same....
the end user shouldn't know about the `returnValues`, nor that it is an event